pkg/storage/streams/store.go

```diff
@@ -97,6 +97,18 @@ func NewStreamStore(segments segments.Store, segmentSize int64, rootKey string,
 func (s *streamStore) Put(ctx context.Context, path paths.Path, data io.Reader, metadata []byte, expiration time.Time) (m Meta, err error) {
 	defer mon.Task()(&ctx)(&err)
 
+	// check if last l/xxx/xx exists?
+	_, err = s.segments.Meta(ctx, path.Prepend("l"))
```

Comment:
After another thought, I think this additional Meta call would have a negative impact on latency. Note that the `s.Delete()` is making the same call too. So why don't we take advantage of this and just call `s.Delete` and check the returned error:
- If it returns with no error then the file already existed and it was deleted successfully. We can proceed with the upload of the new file.
- If it returns `storage.ErrKeyNotFound` then the file did not exist before. We can proceed with the upload.
- If it returns any other error then something wrong happen and we should treat this as error and just return it. No upload will happen.

So we can rework the change in this Put method to look like:

```go
err = s.Delete(ctx, path)
if err != nil && !storage.ErrKeyNotFound.Has(err) {
	// something wrong happened checking for an existing file with the same name
	return Meta{}, err
}
```

Reply:
you are right. No need to do twice. Only thing I did was I just logged an zap error just to give us a reason why we failed. Modified as suggested.

Reply:
If we are returning the error, usually we don't log it at the same time. We leave this privilege to the caller.

Reply:
will remove the error log.